packages/@lwc/engine-server/src/serializer.ts

```diff
@@ -46,20 +46,29 @@ function serializeShadowRoot(shadowRoot: HostShadowRoot): string {
 
 export function serializeElement(element: HostElement): string {
     let output = '';
-    const { name } = element;
+
+    const { name, namespace } = element;
+    const isForeignElement = namespace !== HTML_NAMESPACE;
+    const hasChildren = element.children.length > 0;
 
     const attrs = element.attributes.length ? ` ${serializeAttributes(element.attributes)}` : '';
-    const children = serializeChildNodes(element.children);
 
-    output += `<${name}${attrs}>`;
+    output += `<${name}${attrs}`;
+
+    if (isForeignElement && !hasChildren) {
```

Comment:
TIL that foreign elements don't support shadow root. (E.g. you can't `attachShadow()` on a `<path>`.) Might be good to add a note here.